google/cloud/spanner_dbapi/connection.py

```diff
@@ -104,6 +108,25 @@ def _release_session(self):
         self.database._pool.put(self._session)
         self._session = None
 
+    def retry_transaction(self):
+        """Retry the aborted transaction.
+
+        All the statements executed in the transaction
+        will be re-executed. Results checksums of the original
+        statements and the retried ones will be compared.
+
+        :raises: :class:`google.api_core.exceptions.Aborted`
+            If results checksum of the retried statement is
+            not equal to the checksum of the original one.
+        """
+        for statement in self._statements:
+            res_iter, retried_checksum = self.run_statement(
+                statement, retried=True
+            )
+            for res in res_iter:
+                retried_checksum.consume_result(res)
+                _compare_checksums(statement["checksum"], retried_checksum)
```

Comment:
Here the lengths of the checksums will first be compared. If they are not equal, checksums themselves will not be compared.